WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs

```diff
@@ -1321,357 +1327,332 @@ private async Task<JObject> HandleMergeBranchActionAsync(DataRow dataRowWithSett
-                            fileId = itemId;
-                            originalFileId = fileId;
-                            sqlParameters["fileId"] = fileId;
+                            actionData.FileIdOriginal = actionData.ObjectIdOriginal;
+                            actionData.FileIdMapped = actionData.ObjectIdMapped;
 
-                            if (!filesCache.TryGetValue(tablePrefix, out var listOfFiles))
+                            // If we have a table prefix, then we need to check if it's a valid link type.
+                            if (Int32.TryParse(tablePrefix.TrimEnd('_'), out var linkTypeFromTablePrefix) && allLinkTypeSettings.Any(t => t.UseDedicatedTable && t.Type == linkTypeFromTablePrefix))
                             {
-                                listOfFiles = [];
-                                filesCache.Add(tablePrefix, listOfFiles);
+                                actionData.LinkType = linkTypeFromTablePrefix;
                             }
 
-                            var fileData = listOfFiles.FirstOrDefault(file => file.Id == fileId);
-                            if (fileData != null)
-                            {
-                                if (fileData.IsDeleted)
-                                {
-                                    historyItemsSynchronised.Add(historyId);
-                                    itemsProcessed++;
-                                    await UpdateProgressInQueue(databaseConnection, queueId, itemsProcessed);
-                                    continue;
-                                }
-
-                                if (fileData.ItemId.HasValue && fileData.LinkId.HasValue)
-                                {
-                                    itemId = fileData.ItemId.Value;
-                                    originalItemId = itemId;
-                                    linkId = fileData.LinkId.Value;
-                                    originalLinkId = linkId;
-                                    break;
-                                }
-                            }
-                            else
-                            {
-                                fileData = new BranchMergeFileCacheModel
-                                {
-                                    Id = fileId.Value
-                                };
-                                listOfFiles.Add(fileData);
-                            }
+                            var fileData = await GetFileDataAsync(actionData.FileIdOriginal, sqlParameters, tableName, tablePrefix, branchConnection, filesCache, actionData);
+                            actionData.ItemIdOriginal = fileData.ItemId ?? 0;
+                            actionData.ItemIdMapped = fileData.ItemId ?? 0;
+                            actionData.LinkIdOriginal = fileData.LinkId ?? 0;
+                            actionData.LinkIdMapped = fileData.LinkId ?? 0;
```

Comment:
Is it correct both properties use the same value?

Reply:
Yes that is correct; We want to make sure that we always have a mapped ID. So we always fill the original ID there first, then later we check if that has a mapped ID and if yes, we change it. If it doesn't have a mapped ID, then it will stay the same as the original ID. This way we can just use the mapped ID whenever we need to use that ID in the production database and we're sure it always has a value. See lines 1720+ in this class where we overwrite these values with the actual mapped IDs, if they exist.